### PR TITLE
Use DLT_NULL when writing a pcap file using the Loopback layer

### LINKED ISSUE #3647: Wrong byte order for loopback pcaps
### Brief description

The pcap file format supports two different link-layer header types for loopback: NULL (0), in which the header is in host byte order, and LOOP (108), in which the header is in network byte order.

When scapy reads loopback pcaps, it reads the header in host byte order whether the header type is 0 or 108. When scapy writes loopback pcaps, it stores 108 as the header type but stores the header in host byte order anyway.

This results in Wireshark and other tools saying "Family: Unknown (33554432)" and not decoding the IP packet inside. The code to read pcaps is broken in the same way, so scapy can read its own malformed pcaps but not correctly-formatted ones of type 108.

### Scapy version

2.4.5rc1.dev259

### Python version

3.6.8

### Operating system

RHEL 7.9

### Additional environment information

_No response_

### How to reproduce

Do `wrpcap('loopbackbug.pcap', [Loopback()/IP()/ICMP()])` in scapy and then `tshark -Vr loopbackbug.pcap` from the shell.

### Actual result

It will print "Family: Unknown (33554432)" and not be able to decode the IP or ICMP headers.

### Expected result

It should print "Family: IP (2)" and then decode the IP and ICMP headers.

### Related resources

https://www.tcpdump.org/linktypes.html documents each link header type.

### PR DESCRIPTION
**Pull request by guedou**
This PR fixes #3647 and adds a corresponding test.

### CODE REVIEW

**Comment by josephcsible** [comment]
This doesn't fix the part of #3647 that says "When scapy reads loopback pcaps, it reads the header in host byte order whether the header type is 0 or 108." Please remove the phrase "fixes #3647" from your commit and PR description so that the issue doesn't get auto-closed before it's entirely fixed.

**Comment by guedou** [comment]
Could you share snippets of broken behaviors? That greatly simplifies our work. Thanks.

File: scapy/layers/l2.py [hunk 1]
```diff
@@ -720,7 +719,7 @@ class Dot1AD(Dot1Q):
 conf.l2types.register(DLT_LINUX_SLL2, CookedLinuxV2)
 conf.l2types.register(DLT_ETHERNET_MPACKET, MPacketPreamble)
 conf.l2types.register_num2layer(DLT_LINUX_IRDA, CookedLinux)
-conf.l2types.register(DLT_LOOP, Loopback)
+conf.l2types.register(DLT_NULL, Loopback)
 conf.l2types.register_num2layer(DLT_NULL, Loopback)
```
Comment (gpotter2) [inline scapy/layers/l2.py:723]: Doesn't it get redundant with this?
Should we have
`conf.l2typed.register_num2layer(DLT_LOOP, Loopback)` instead for compatibility?

Comment (guedou) [reply scapy/layers/l2.py]: You are right.

**Review by gpotter2 — APPROVED** [review]

**Pull request merged**